src/mono/mono/component/debugger-agent.c

```diff
@@ -8282,6 +8328,21 @@ field_commands (int command, guint8 *p, guint8 *end, Buffer *buf)
 		}
 		break;
 	}
+	case MDBGPROT_CMD_FIELD_GET_TOKEN_AND_TYPE: {
+		MonoClassField *f = decode_fieldid (p, &p, end, &domain, &err);
+		if (G_UNLIKELY (!f->type)) {
+			ERROR_DECL(field_error);
+			mono_field_resolve_type (f, field_error);
+			mono_error_cleanup (field_error);
+			if (!f->type)
+				/* ERROR */;
```

Comment:
This should actually error out - otherwise you'll dereference a null pointer on line 8341

```suggestion
				return ERR_INVALID_ARGUMENT;
```

"invalid argument" is not right.  do we have a  "bad object" or "bad class" error code?